sdk/core/azure-core/src/http/bodyStream.cpp

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifdef POSIX
+#include <unistd.h>
+#endif
+
+#ifdef WINDOWS
+#define WIN32_LEAN_AND_MEAN
+#define NOMINMAX
+#include <Windows.h>
+#endif // Windows
+
+#include <algorithm>
+#include <context.hpp>
+#include <cstdint>
+#include <cstdio>
+#include <cstring>
+#include <http/bodyStream.hpp>
+#include <memory>
+#include <vector>
+
+using namespace Azure::Core::Http;
+
+BodyStream::~BodyStream() {}
+
+// Keep reading until buffer is all fill out of the end of stream content is reached
+int64_t BodyStream::ReadToCount(Context& context, BodyStream& body, uint8_t* buffer, int64_t count)
+{
+  int64_t totalRead = 0;
+
+  for (;;)
+  {
+    int64_t readBytes = body.Read(context, buffer + totalRead, count - totalRead);
+    totalRead += readBytes;
+    // Reach all of buffer size
+    if (totalRead == count || readBytes == 0)
+    {
+      return totalRead;
+    }
+  }
+}
+
+std::unique_ptr<std::vector<uint8_t>> BodyStream::ReadToEnd(Context& context, BodyStream& body)
+{
+  constexpr int64_t chunkSize = 1024 * 8;
```

Comment:
Magic numbers should be documented. Ideally, this is a changeable size for the client.

Reply:
@katmsft Let's be less picky about these trivial stuff and focus on the ability to function properly at this stage.